base/math.jl

```diff
@@ -910,14 +910,13 @@ end
     end
     z
 end
-@inline function ^(x::Float32, y::Float32)
-    z = ccall("llvm.pow.f32", llvmcall, Float32, (Float32, Float32), x, y)
-    if isnan(z) & !isnan(x+y)
-        throw_exp_domainerror(x)
-    end
-    z
+@inline function ^(x::T, y::T) where T <: Union{Float16, Float32}
+    x==0 && return T(y==0)
```

Comment:
What do we want for the following cases?

- 0.0^0.0
- 0.0^-1.0
- 0.0^NaN
I've gone with 1 for 0^0, and otherwise 0., which I think is reasonable.

Reply:
Looks like this is what we do for `Float64`. As long as it's consistent, this seems fine to me.

Reply:
Ah, wait. We do have:
```julia
julia> 0.0 ^ NaN
NaN
```
so this should probably be matched.

Reply:
I'm somewhat unclear about if this is intended given that `0.0^negative` is clearly wrong. It's reasonable behavior, but not the only reasonable behavior.

Reply:
Also worth noting is that `1.0^NaN` currently gives `1.0`, and `NaN^0.0` gives `1.0`. Having similar behavior for `0.0` seems reasonable.

Reply:
IEEE Std 754-2019 

6.2 Operations with NaNs
Under default exception handling, any operation signaling an invalid operation exception and for which a floating-point result is to be delivered, except as stated otherwise, shall deliver a quiet NaN.

6.2.3 NaN propagation
An operation that propagates a NaN operand to its result and has a single NaN as an input should produce a NaN with the payload of the input NaN if representable in the destination format.

seems to me that `float^NaN should == NaN^float == NaN`